src/test/java/org/mifos/integrationtest/cucumber/stepdef/VoucherManagementStepDef.java

```diff
@@ -0,0 +1,424 @@
+    private static  String callbackBody;
+    private static String serialNumber;
+    private static String voucherNumber;
+    private static String cancelVoucherBody;
+    private static String suspendVoucherBody;
+    private static String registeringInstitutionId = "SocialWelfare";
+    @Autowired
+    MockServerStepDef mockServerStepDef;
+
+
+    @Given("I can create an VoucherRequestDTO for voucher creation")
+    public void iCreateAnIdentityMapperDTOForRegisterBeneficiary() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("{\n");
+        sb.append("    \"requestID\": \"").append(generateUniqueNumber(14)).append("\",\n");
+        sb.append("    \"batchID\": \"").append(generateUniqueNumber(10)).append("\",\n");
+        sb.append("    \"voucherInstructions\": [\n");
+        sb.append("        {\n");
+        sb.append("            \"instructionID\": \"").append(generateUniqueNumber(14)).append("\",\n");
+        sb.append("            \"groupCode\": \"021\",\n");
+        sb.append("            \"currency\": \"SGD\",\n");
+        sb.append("            \"amount\": 5000,\n");
+        sb.append("            \"payeeFunctionalID\": \"63310590322288932682\",\n");
+        sb.append("            \"narration\": \"Social Support Payment for the Month of January 2023\"\n");
+        sb.append("        },\n");
+        sb.append("        {\n");
+        sb.append("            \"instructionID\": \"").append(generateUniqueNumber(14)).append("\",\n");
+        sb.append("            \"groupCode\": \"021\",\n");
+        sb.append("            \"currency\": \"SGD\",\n");
+        sb.append("            \"amount\": 3000,\n");
+        sb.append("            \"payeeFunctionalID\": \"08187379158030059976\",\n");
+        sb.append("            \"narration\": \"Social Support Payment for the Month of January 2023\"\n");
+        sb.append("        }\n");
+        sb.append("    ]\n");
+        sb.append("}");
+        createVoucherBody = sb.toString();
+    }
+
+    @When("I call the create voucher API with expected status of {int} and stub {string}")
+    public void iCallTheVoucherCreateAPIWithExpectedStatusOf(int expectedStatus, String stub) {
+        BaseStepDef.response = RestAssured.given()
+                .header("Content-Type", "application/json")
+                .header("X-CallbackURL",mockServer.getBaseUri()+stub)
+                .header("registeringInstitutionId", registeringInstitutionId)
+                .baseUri(voucherManagementConfig.voucherManagementContactPoint)
+                .body(createVoucherBody)
+                .expect()
+                .spec(new ResponseSpecBuilder().expectStatusCode(expectedStatus).build())
+                .when()
+                .post(voucherManagementConfig.createVoucherEndpoint)
+                .andReturn().asString();
+
+
+        logger.info("Identity Mapper Response: {}", BaseStepDef.response);
+        try {
+            Thread.sleep(10000);
```

Comment:
this is not a good practice to add delays, together it can take forever to finish the integration test run. find a better way.